src/shared/seccomp-util.c

```diff
@@ -1486,10 +1486,16 @@ static int add_seccomp_syscall_filter(scmp_filter_ctx seccomp,
                                       uint32_t arch,
                                       int nr,
                                       unsigned arg_cnt,
-                                      const struct scmp_arg_cmp arg) {
+                                      const struct scmp_arg_cmp arg,
+                                      int multiplexed) {
         int r;
 
-        r = seccomp_rule_add_exact(seccomp, SCMP_ACT_ERRNO(EPERM), nr, arg_cnt, arg);
+        /* While exact is preferred multiplexed calls are known to fail on _exact */
+        if (multiplexed)
+            r = seccomp_rule_add(seccomp, SCMP_ACT_ERRNO(EPERM), nr, arg_cnt, arg);
+        else
+            r = seccomp_rule_add_exact(seccomp, SCMP_ACT_ERRNO(EPERM), nr, arg_cnt, arg);
```

Comment:
we stopped using the non-exact versions a long to ago since you never knew what you#d get, when used interchangably in whitelists and blacklists... i'd prefer if we can just deal with this issue in our own code.

indentation is off btw, please stick to 8ch, see https://systemd.io/CODING_STYLE

Reply:
(also, we use `bool` for booleans, except in public, exposed APIs, which this is not)

Reply:
Hi @poettering,
indention already fixed a minute ago after @keszybz pointing it out.

I'll carry the preference to stick to `_excat` calls to the discussion in libseccomp project.
If there is a way to stick to `_exact` that works I'm happy to use that and will refresh this pull request.

But ahead of time if there isn't a way what would we want to do about s390 and x86 then instead. We could obviously just ignore the retval in those cases, but that means less protection. Do you have any alternative ideas in mind already?

Reply:
I asked a similar question before and the answer I just got was

> I would suggest using seccomp_rule_add(...) as it will handle all the architecture oddities for you; in fact for 99% of the cases out there seccomp_rule_add(...) is preferred over seccomp_rule_add_exact(...).

To me that seems that like only way to go unless we switch to more drastic changes (after all TBH who cares about 31/32bit - lets maybe just remove their shmat call, _/me runs away ..._)

I asked @pcmoore to come over and participate in the discussion here directly.

Reply:
> (also, we use `bool` for booleans, except in public, exposed APIs, which this is not)

Fixed as well, thanks